utils/src/config_setting.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_export]
+macro_rules! make_config_setting {
+    ($name: ident, $tp: ty, $default_value: expr) => {
+        #[derive(Debug, Clone)]
+        pub struct $name {
+            value: $tp,
+        }
+
+        impl From<$tp> for $name {
+            fn from(v: $tp) -> Self {
+                Self { value: v }
+            }
+        }
+
+        impl From<$name> for $tp {
+            fn from(v: $name) -> Self {
+                v.value
+            }
+        }
+
+        impl From<Option<$tp>> for $name {
+            fn from(v: Option<$tp>) -> Self {
+                Self {
+                    value: v.unwrap_or(Self::default().value),
+                }
+            }
+        }
```

Comment:
I think this is just `Option::unwrap_or_default`. Could that be used instead of a custom `From` impl to be more explicit?

Reply:
Unfortunately, that doesn't work, because the default would be for v of type `Option<$tp>`, whose default is not really what we're looking for.

Reply:
Hmm. I see. The impl looks a bit awkward though. Where is it used?

Reply:
This is basically the de-facto conversion method from "user didn't supply a value" to "default value".

Reply:
I guess my question is why not use conversion from `Option<$name>` rather than `Option<$tp>` in that case, and use the more expressive wrapper types in the configuration files / command lines as well. Is there an issue with that? There might be something I'm missing.

Reply:
Because this type has nothing to do with serialization. The only purpose of this type is to have a well-defined way of interpreting input options. The serialization of `$tp` comes from command line arguments, which have their serialization defined with them. 

If you think about it, it'll become a circular problem. You add serialization to this type and use it in command line arguments, and we're back to the semi-predicate problem, where we can't distinguish between None and default. I don't know, maybe I'm missing something.